pkg/binding/bindingpolicy.go

```diff
@@ -101,6 +142,23 @@ func (c *Controller) syncBindingPolicy(ctx context.Context, bindingPolicyName st
 	return c.deleteResolutionForBindingPolicy(ctx, bindingPolicyName)
 }
 
+func (c *Controller) identifyMissingStatusCollector(bp *v1alpha1.BindingPolicy) (string, error) {
+	for _, clause := range bp.Spec.Downsync {
+		if clause.DownsyncModulation.StatusCollectors == nil {
+			continue
+		}
```

Comment:
https://go.dev/ref/spec#For_range, case 1, explicitly allows `range` over `nil` and it does what you would expect.